lib/semantics/check-omp-structure.cc

```diff
@@ -404,6 +412,15 @@ void OmpStructureChecker::Enter(const parser::OpenMPBlockConstruct &x) {
         OmpClause::DEVICE, OmpClause::DEFAULTMAP, OmpClause::NOWAIT};
     SetContextAllowedOnce(allowedOnce);
   } break;
+  case parser::OmpBlockDirective::Directive::TargetData: {
+    PushContext(beginDir.source, OmpDirective::TARGET_DATA);
+    OmpClauseSet allowed{
+        OmpClause::IF, OmpClause::MAP, OmpClause::USE_DEVICE_PTR};
+    SetContextAllowed(allowed);
+    OmpClauseSet allowedOnce{OmpClause::DEVICE};
+    SetContextAllowedOnce(allowedOnce);
```

Comment:
Could be similar to the next line `SetContextAllowedOnce(OmpClauseSet{OmpClause::DEVICE});`.

Reply:
There are many places in this file where there is a named temporary like this. It would be good for them to be consistent.

I think you could just write these like:
`SetContextAllowedOnce({OmpClause::DEVICE});`

Reply:
Right.

Reply:
I think when there's only one we should consistently not create a temporary, and more than one we should as it's clearer. But I don't have strong feelings either way so I'll change it to whatever we think is best